src/test/log_console_test.cpp

```diff
@@ -21,6 +21,24 @@
 
 namespace ppx {
 
+class LogStaticTest : public ::testing::Test
+{
+protected:
+    void SetUp() override
+    {
+        // Some other unrelated tests might have run before the logging tests and
+        // already initialized logging. Since we share global state and run all
+        // tests in a single process, we need to shut down any existing logging.
+        // If the logging was not initialized this operation is a no-op.
+        Log::Shutdown();
+    }
+
+    void TearDown() override
+    {
+        Log::Shutdown();
```

Comment:
I think we don't actually need to call Log::Shutdown() since the tests do it themselves, and in one case it is even part of the test. Though I guess right now it is pretty harmless to call it again.

Reply:
I can remove it if you want. Even if it is not needed in the current test, I wanted to follow the pattern of `LogTest` where we guarantee that the log is shutdown after every test. I don't think it is a big deal either way. Tests are assuming the Log has not been shutdown, so it might be fair to not shut it down.

Reply:
Yeah, I don't really have a strong preference either way, as you said there is an argument for either. Since it's currently this way, lgtm